rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/implicits.scala

```diff
@@ -26,15 +27,16 @@ object implicits {
     }
 
   implicit val matchListQuote
-    : StorageMatch[BindPattern, ListChannelWithRandom, ListChannelWithRandom] =
-    new StorageMatch[BindPattern, ListChannelWithRandom, ListChannelWithRandom] {
+    : StorageMatch[BindPattern, OutOfPhloError.type, ListChannelWithRandom, ListChannelWithRandom] =
```

Comment:
Why not `InterpreterError`? Maybe we should introduce `RholangRuntimeError` or something similar?

Reply:
Because it is the only error that matching can fail with.

Reply:
Can anything aside from the matcher fail with an OutOfPhloError?